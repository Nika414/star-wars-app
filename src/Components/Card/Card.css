.card {
    background: #F0F0F0;
    border-radius: 8px;
    width: 351px;
    min-width: 300px;
    margin: 0 10px;
    height: 137px;
    filter: drop-shadow(4px 4px 4px rgba(33, 33, 33, 0.1));
    cursor: pointer;
}

.card__title {
    font-style: normal;
    font-weight: 700;
    font-size: 18px;
    line-height: 21px;
    color: #212121;
    text-shadow: 4px 4px 4px rgba(33, 33, 33, 0.1);
    margin: 5px 0 19px 7.98%;
}

.card__card-tag {
    min-width: 74px;
    width: fit-content;
    height: 16px;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    color: #212121;
    box-shadow: inset 0px -2px 0px rgba(0, 0, 0, 0.18);
    border-radius: 11px;
    padding: 0 11px;
    box-sizing: border-box;
}

.card__info-container {
    display: flex;
    gap: 13px;
    margin: 0 0 10px 7.98%;

}

.card__info-value {
    border: 3px solid #212121;
    box-sizing: border-box;
    border-radius: 50%;
    min-width: 38px;
    width: fit-content;
    height: 36px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-style: normal;
    font-weight: 400;
    font-size: 15px;
    line-height: 18px;
    text-align: center;
    color: #212121;
    margin: 0;
    margin-bottom: 6px;
}

.card__info-name {
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 14px;
    text-align: center;
    color: #4B4B4B;
}

.card__info-item {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.card__info-item_place_popup {
    background-color: white;
    height: 70px;
    width: 83px;
    border-radius: 8px;
    margin-right: 24px;
    justify-content: center;
}

.card__tags-container {
    display: flex;
    gap: 13px;
    margin: 0 0 12px 7.98%;
}

.card__info-item_hidden {
    display: none;
}