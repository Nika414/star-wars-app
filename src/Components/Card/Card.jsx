import { v4 as uuid } from 'uuid';
import CardTag from '../CardTag';
import CardInfo from '../CardInfo';

export default function Card({ card, onClick }) {
  function handleClick() {
    onClick(card);
  }

  return (
    <div onClick={handleClick} className="card" role="presentation">
      <h3 className="card__title">{card.name}</h3>
      <div className="card__info-container">
        <CardInfo name="height" value={card.height} />
        <CardInfo name="mass" value={card.mass} />
      </div>
      <div className="card__tags-container">
        {card.tags.map((tag) => <CardTag key={uuid()} info={tag.info} color={tag.color} />)}
      </div>
    </div>
  );
}
