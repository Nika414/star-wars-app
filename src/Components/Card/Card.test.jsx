/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Card from './Card';

const data = {
  name: 'Luke Skywalker',
  height: '172',
  mass: '77',
  hair_color: 'blond',
  skin_color: 'fair',
  eye_color: 'blue',
  birth_year: '19BBY',
  gender: 'male',
  homeworld: 'https://swapi.dev/api/planets/1/',
  films: [
    'https://swapi.dev/api/films/2/',
    'https://swapi.dev/api/films/6/',
    'https://swapi.dev/api/films/3/',
    'https://swapi.dev/api/films/1/',
    'https://swapi.dev/api/films/7/',
  ],
  species: [
    'https://swapi.dev/api/species/1/',
  ],
  vehicles: [
    'https://swapi.dev/api/vehicles/14/',
    'https://swapi.dev/api/vehicles/30/',
  ],
  starships: [
    'https://swapi.dev/api/starships/12/',
    'https://swapi.dev/api/starships/22/',
  ],
  created: '2014-12-09T13:50:51.644000Z',
  edited: '2014-12-20T21:17:56.891000Z',
  url: 'https://swapi.dev/api/people/1/',
  tags: [
    {
      info: '112BBY',
      color: '#07D6F2',
    },
  ],
};

const onClick = jest.fn();

describe('Card component', () => {
  it('Card renders', () => {
    render(<Card card={data} />);

    expect(screen.getByText('Luke Skywalker')).toBeInTheDocument();
  });
  it('Card snapshot', () => {
    const card = render(<Card card={data} />);
    expect(card).toMatchSnapshot();
  });
  it('onChange works', () => {
    render(<Card card={data} onClick={onClick} />);
    field.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
