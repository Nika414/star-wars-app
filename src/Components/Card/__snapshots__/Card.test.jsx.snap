// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        role="presentation"
      >
        <h3
          class="card__title"
        >
          Luke Skywalker
        </h3>
        <div
          class="card__info-container"
        >
          <div
            class="card__info-item  undefined"
          >
            <div
              class="card__info-value"
            >
              172
            </div>
            <span
              class="card__info-name"
            >
              height
            </span>
          </div>
          <div
            class="card__info-item  undefined"
          >
            <div
              class="card__info-value"
            >
              77
            </div>
            <span
              class="card__info-name"
            >
              mass
            </span>
          </div>
        </div>
        <div
          class="card__tags-container"
        >
          <div
            class="card__card-tag"
            style="background-color: rgb(7, 214, 242);"
          >
            112BBY
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      role="presentation"
    >
      <h3
        class="card__title"
      >
        Luke Skywalker
      </h3>
      <div
        class="card__info-container"
      >
        <div
          class="card__info-item  undefined"
        >
          <div
            class="card__info-value"
          >
            172
          </div>
          <span
            class="card__info-name"
          >
            height
          </span>
        </div>
        <div
          class="card__info-item  undefined"
        >
          <div
            class="card__info-value"
          >
            77
          </div>
          <span
            class="card__info-name"
          >
            mass
          </span>
        </div>
      </div>
      <div
        class="card__tags-container"
      >
        <div
          class="card__card-tag"
          style="background-color: rgb(7, 214, 242);"
        >
          112BBY
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
